rs/boundary_node/anonymization/backend/src/lib.rs

```diff
@@ -1,2 +1,118 @@
-// I put this here for CI, will remove on the next PR
-pub const TMP: u32 = 0;
+use std::{cell::RefCell, time::Duration};
+
+use anonymization_interface::{
+    self as ifc, InitArg, QueryResponse, RegisterResponse, SubmitResponse,
+};
+use candid::Principal;
+use ic_cdk::{id, spawn};
+use ic_cdk_timers::set_timer_interval;
+use ic_stable_structures::{
+    memory_manager::{MemoryId, MemoryManager, VirtualMemory},
+    DefaultMemoryImpl, StableBTreeMap,
+};
+use lazy_static::lazy_static;
+use registry::{Client, List};
+
+mod registry;
+
+type Memory = VirtualMemory<DefaultMemoryImpl>;
+
+type StableMap<K, V> = StableBTreeMap<K, V, Memory>;
+type StableSet<T> = StableMap<T, ()>;
+
+thread_local! {
+    static MEMORY_MANAGER: RefCell<MemoryManager<DefaultMemoryImpl>> =
+        RefCell::new(MemoryManager::init(DefaultMemoryImpl::default()));
+}
+
+const REGISTRY_CANISTER_ID: &str = "bnz7o-iuaaa-aaaaa-qaaaa-cai";
+
+const MEMORY_ID_ALLOWED_PRINCIPALS: u8 = 0;
+
+lazy_static! {
+    static ref REGISTRY_LISTER: Box<dyn List> = {
+        let cid =
+            Principal::from_text(REGISTRY_CANISTER_ID).expect("failed to construct principal");
+
+        let v = Client::new(cid);
+        Box::new(v)
+    };
+}
+
+thread_local! {
+    static ALLOWED_PRINCIPALS: RefCell<StableSet<Principal>> = RefCell::new(
+        StableSet::init(
+            MEMORY_MANAGER.with(|m| m.borrow().get(MemoryId::new(MEMORY_ID_ALLOWED_PRINCIPALS))),
+        )
+    );
+}
+
+// Timers
+
+fn timers() {
+    // ACLs
+    set_timer_interval(Duration::from_secs(10), || {
```

Comment:
Suggestion: move hard-coded value to `timers()` or make it const. You could also put it to `InitArg`